code/modules/organs/organ_internal.dm

```diff
@@ -122,9 +118,25 @@
 	var/datum/limb/parent = owner.get_limb(parent_limb)
 	if (!silent)
 		owner.custom_pain("Something inside your [parent.display_name] hurts a lot.", 1)
+	set_organ_status()
+
+/// Set the correct organ state
+/datum/internal_organ/proc/set_organ_status()
+	if(damage > min_broken_damage)
+		if(organ_status != ORGAN_BROKEN)
+			organ_status = ORGAN_BROKEN
+		return TRUE
+	if(damage > min_bruised_damage)
+		if(organ_status != ORGAN_BRUISED)
+			organ_status = ORGAN_BRUISED
+		return TRUE
+	if(organ_status != ORGAN_HEALTHY)
+		organ_status = ORGAN_HEALTHY
+		return TRUE
 
 /datum/internal_organ/proc/heal_organ_damage(amount)
 	damage = max(damage - amount, 0)
+	set_organ_status()
```

Comment:
Do this when taking damage too

Reply:
oh i forgot

Reply:
uh, but i was doing it haha, i didn't forget